public/app/features/query/state/PanelQueryRunner.ts

```diff
@@ -111,26 +113,35 @@ export class PanelQueryRunner {
       return of(snapshotPanelData);
     }
 
-    return this.subject.pipe(
-      mergeMap((data: PanelData) => {
+    const panel = this.dataConfigSource as unknown as PanelModel;
+    let dashData: Observable<DashboardQueryRunnerResult>;
+
+    try {
+      dashData = getDashboardQueryRunner().getResult(panel.id);
```

Comment:
Isn't it a bit weird that we call `DashboardQueryRunner` from `PanelQueryRunner`?

Reply:
Kinda of but also not really – the `DashboardQueryRunner` will fetch data that is relevant to be distributed to panels.

The current feature to get the alert rule state for panels uses the same paradigm where the dashboard query runner fetches the alert states and the panels get the results only they are interested in.